### PR TITLE
dmd.toir: Merge 32 and 64bit intrinsic arrays

### PR DESCRIPTION
**Pull request by ibuclaw**
As for the intrinsic names, there's only 4 places where they differ between the two.

### CODE REVIEW

File: src/dmd/toir.d [hunk 1]
```diff
@@ -444,86 +427,7 @@ int intrinsic_op(FuncDeclaration fd)
...
-        "8volatile12volatileLoadFNbNiNfPmZm",
-        "8volatile12volatileLoadFNbNiNfPtZt",
-        // volatileStore
-        "8volatile13volatileStoreFNbNiNfPhhZv",
-        "8volatile13volatileStoreFNbNiNfPkkZv",
-        "8volatile13volatileStoreFNbNiNfPmmZv",
-        "8volatile13volatileStoreFNbNiNfPttZv",
-    ];
-    __gshared immutable char*[62] core_namearray64 =
+    immutable char*[62] core_namearray =
```
Comment (rainers) [inline src/dmd/toir.d]: Are you sure you want to initialize these arrays with every function call? They are no longer static, as you can use runtime variables during initialization.

Comment (ibuclaw) [reply src/dmd/toir.d]: We can add a guard just to be safe.

**Comment by ibuclaw** [comment]
Looks like @thewilsonator had a better idea in #10656.  :-o

File: src/dmd/toir.d [hunk 2]
```diff
@@ -395,40 +395,28 @@ int intrinsic_op(FuncDeclaration fd)
...
-        "4math4rintFNaNbNiNfeZe",
-        "4math4sqrtFNaNbNiNfdZd",
-        "4math4sqrtFNaNbNiNfeZe",
-        "4math4sqrtFNaNbNiNffZf",
-        "4math4yl2xFNaNbNiNfeeZe",
-        "4math5ldexpFNaNbNiNfeiZe",
-        "4math6rndtolFNaNbNiNfeZl",
-        "4math6yl2xp1FNaNbNiNfeeZe",
-    ];
+    __gshared immutable(char)*[11] std_namearray;
```
Comment (rainers) [inline src/dmd/toir.d]: No need for runtime initialization here, just use `static immutable`.

File: src/dmd/toir.d [hunk 3]
```diff
@@ -444,164 +432,90 @@ int intrinsic_op(FuncDeclaration fd)
...
-        // volatileStore
-        "8volatile13volatileStoreFNbNiNfPhhZv",
-        "8volatile13volatileStoreFNbNiNfPkkZv",
-        "8volatile13volatileStoreFNbNiNfPmmZv",
-        "8volatile13volatileStoreFNbNiNfPttZv",
-    ];
+    __gshared immutable(char)*[62] core_namearray;
+    if (init_guard < 2)
+    {
+        core_namearray = [
```
Comment (rainers) [inline src/dmd/toir.d]: I think this assignment will cause the an allocation of the array literal. Happens only once, but maybe better to just use static initialization and then patching the 4 entries?

**Pull request closed without merging**